gcp_variant_transforms/transforms/merge_header_definitions_test.py

```diff
@@ -0,0 +1,236 @@
+    self.assertEqual(
+        format_definitions['NS'][Definition(1, 'Float')],
+        list(['file2']))
+
+  def test_merge_header_definitions_save_five_copies(self):
+    lines_1 = [
+        '##fileformat=VCFv4.2\n',
+        '##INFO=<ID=NS,Number=1,Type=Float,Description="Number samples">\n',
+        '#CHROM  POS ID  REF ALT QUAL  FILTER  INFO  FORMAT  Sample1 Sample2\n']
+    lines_2 = [
+        '##fileformat=VCFv4.2\n',
+        '##INFO=<ID=NS,Number=1,Type=Integer,Description="Number samples">\n',
+        '#CHROM  POS ID  REF ALT QUAL  FILTER  INFO  FORMAT  Sample3\n']
+
+    vcf_reader_1 = vcf.Reader(fsock=iter(lines_1))
+    vcf_reader_2 = vcf.Reader(fsock=iter(lines_2))
+    headers_1 = self._get_vcf_header_from_reader(vcf_reader_1, 'file1')
+    headers_2 = self._get_vcf_header_from_reader(vcf_reader_1, 'file2')
+    headers_3 = self._get_vcf_header_from_reader(vcf_reader_1, 'file3')
+    headers_4 = self._get_vcf_header_from_reader(vcf_reader_1, 'file4')
+    headers_5 = self._get_vcf_header_from_reader(vcf_reader_1, 'file5')
+    headers_6 = self._get_vcf_header_from_reader(vcf_reader_1, 'file6')
+    headers_7 = self._get_vcf_header_from_reader(vcf_reader_2, 'file7')
+    combiner_fn = self._get_definitions_combiner_fn()
+
+    merged_headers = combiner_fn.create_accumulator()
+    merged_headers = combiner_fn.add_input(merged_headers, headers_1)
+    merged_headers = combiner_fn.add_input(merged_headers, headers_2)
+    merged_headers = combiner_fn.add_input(merged_headers, headers_3)
+    merged_headers = combiner_fn.add_input(merged_headers, headers_4)
+    merged_headers = combiner_fn.add_input(merged_headers, headers_5)
+    merged_headers = combiner_fn.add_input(merged_headers, headers_6)
+    merged_headers = combiner_fn.add_input(merged_headers, headers_7)
+
+    definitions = combiner_fn.extract_output(merged_headers)
+    info_definitions = definitions.infos
+    format_definitions = definitions.formats
+    self.assertEqual(info_definitions.keys(), ['NS'])
+    self.assertEqual(
+        info_definitions['NS'][Definition(1, 'Float')],
+        list(['file1', 'file2', 'file3', 'file4', 'file5']))
+    self.assertEqual(
+        info_definitions['NS'][Definition(1, 'Integer')],
+        list(['file7']))
+    self.assertEqual(len(format_definitions), 0)
+
+  def test_combine_pipeline(self):
+    vcf_reader_1 = vcf.Reader(fsock=iter(FILE_1_LINES))
+    vcf_reader_2 = vcf.Reader(fsock=iter(FILE_2_LINES))
+    headers_1 = self._get_vcf_header_from_reader(vcf_reader_1, 'file1')
+    headers_2 = self._get_vcf_header_from_reader(vcf_reader_2, 'file2')
+
+    conflicts_merger = merge_header_definitions._DefinitionsMerger()
+    expected = merge_header_definitions.VcfHeaderDefinitions()
+    conflicts_merger.merge(
+        expected, merge_header_definitions.VcfHeaderDefinitions(headers_1))
+    conflicts_merger.merge(
+        expected, merge_header_definitions.VcfHeaderDefinitions(headers_2))
+
+    pipeline = TestPipeline()
```

Comment:
@nmousavi : I think there was a TODO at some point to migrate all tests to use the MergeDefinitions transform directly instead of the combine_fn ones?

Reply:
I found the TODO in `merge_headers_tests`. I have updated the test cases to use the pipelines directly for `merge_header_definitions_test`.